torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -479,6 +482,9 @@ class AttributePropagator {
     if (node->kind() != prim::GetAttr) {
       return;
     }
+    if (!node->output()->type()->cast<ClassType>()) {
```

Comment:
Why are we doing this check ? Could you add a comment into the code ?

Reply:
One of my unit test fired because I call a global method? I don't anticipate this to happen in practice. 
I feel this is still fragile because  we expect the first argument of fork to be submodule of the invoked method which is just a convention. 
Anyway freezing will fail if it tries to resolve an attribute that does not exist.  I guess it is okay in case someone wants to get creative and pass the submodule as second argument ...

I will add a comment

Reply:
I think we should still be recursively analyzing the fork even if it's not a forked module - don't we still want to inline interface calls and such ?

Reply:
I agree. I updated my PR to handle in this case